packages/api/src/config.js

```diff
@@ -0,0 +1,321 @@
+  MAILCHIMP_API_KEY: '',
+  LOGTAIL_TOKEN: '',
+  S3_ENDPOINT: 'http://localhost:9095',
+  S3_REGION: 'test',
+  S3_ACCESS_KEY_ID: 'test',
+  S3_SECRET_ACCESS_KEY: 'test',
+  S3_BUCKET_NAME: 'test',
+  PRIVATE_KEY: 'xmbtWjE9eYuAxae9G65lQSkw36HV6H+0LSFq2aKqVwY=',
+}
+
+/**
+ * If the CLUSTER_SERVICE variable is set, the service URL will be resolved from here.
+ *
+ * @type Record<string, string> */
+const CLUSTER_SERVICE_URLS = {
+  IpfsCluster: 'https://nft.storage.ipfscluster.io/api/',
+  IpfsCluster2: 'https://nft2.storage.ipfscluster.io/api/',
+  IpfsCluster3: 'https://nft3.storage.ipfscluster.io/api/',
+}
+
+/**
+ * @param {RuntimeEnvironmentName} env
+ * @returns {boolean} true if the named runtime environment should fallback to values from DEFAULT_CONFIG_VALUES if no config var is present.
+ */
+const allowDefaultConfigValues = (env) => env === 'test' || env === 'dev'
+
+/**
+ * Returns a {@link ServiceConfiguration} object containing the runtime config options for the API service.
+ * Includes anything injected by the environment (secrets, URLs for services we call out to, maintenance flag, etc).
+ *
+ * Loaded from global variables injected by CloudFlare Worker runtime.
+ *
+ * Lazily loaded and cached on first access.
+ */
+export const getServiceConfig = () => {
+  if (_globalConfig) {
+    return _globalConfig
+  }
+  _globalConfig = loadServiceConfig()
+  return _globalConfig
+}
+
+/** @type ServiceConfiguration|undefined */
+let _globalConfig
+
+/**
+ * Load a {@link ServiceConfiguration} from the global environment.
+ *
+ * Exported for testing. See {@link getServiceConfig} for main public accessor.
+ *
+ * Config values are resolved by looking for global variables with the names matching the keys of {@link DEFAULT_CONFIG_VALUES}.
+ *
+ * If no global value is found for a variable, an error will be thrown if the runtimeEnvironment (ENV variable)
+ * is set to a "production like" environment.
+ *
+ * If {@link allowDefaultConfigValues} returns true for the current environment, the value from {@link DEFAULT_CONFIG_VALUES} will be
+ * used if a variable is missing.
+ *
+ * @returns {ServiceConfiguration}
+ */
+export function loadServiceConfig() {
+  const vars = loadConfigVariables()
+  mergeDefaultValuesIfAllowed(vars)
+
+  return serviceConfigFromVariables(vars)
+}
+
+/**
+ * Parse a {@link ServiceConfiguration} out of the given `configVars` map.
+ * @param {Record<string, string>} configVars map of variable names to values.
+ *
+ * Exported for testing. See {@link getServiceConfig} for main public accessor.
+ *
+ * @returns {ServiceConfiguration}
+ */
+export function serviceConfigFromVariables(configVars) {
+  let clusterUrl = configVars.CLUSTER_API_URL
+  if (configVars.CLUSTER_SERVICE) {
+    const serviceUrl = CLUSTER_SERVICE_URLS[configVars.CLUSTER_SERVICE]
+    if (!serviceUrl) {
+      throw new Error(`unknown cluster service: ${configVars.CLUSTER_SERVICE}`)
+    }
+    clusterUrl = serviceUrl
+  }
+
+  return {
```

Comment:
since we are changing things, this object structure always felt wrong for config constants with all the nesting etc... we had reasons for this but not anymore.

my suggestion would be flatten it just like the DEFAULT_CONFIG_VALUES object

Reply:
Also it would be easier to spot config globals, if we could import config and do `config.S3_ENDPOINT` instead of `config.s3.endpoint` or whatever we do now.

Reply:
Sweet, that makes things quite a bit simpler 😄It sounds like the final config object will look just like `DEFAULT_CONFIG_VALUES`, plus a little bit of type validation (string to bool, validating maintenance mode, etc).